pages/lists/[listId]/highlights.tsx

```diff
@@ -0,0 +1,267 @@
+        bearerToken,
+        pathValidator: validateListPath,
+      }),
+      fetchApiData<DBList>({ path: `lists/${listId}`, bearerToken, pathValidator: validateListPath }),
+      fetchApiData<PagedData<DbHighlight>>({
+        path: `lists/${listId}/contributors/highlights?repo=${repo}`,
+        bearerToken,
+        pathValidator: validateListPath,
+      }),
+    ]);
+
+  if (error?.status === 404) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const userId = Number(session?.user.user_metadata.sub);
+
+  return {
+    props: {
+      list,
+      numberOfContributors: data?.meta.itemCount || 0,
+      isOwner: list && list.user_id === userId,
+      highlights: {
+        data: highlights?.data || [],
+        meta: highlights?.meta || {},
+      },
+    },
+  };
+};
+
+const rangeFilterOptions: { label: string; value: string }[] = [
+  {
+    label: "Past week",
+    value: "7",
+  },
+  {
+    label: "Past 30 days",
+    value: "30",
+  },
+  {
+    label: "Past 3 months",
+    value: "90",
+  },
+];
+
+const Highlights = ({ list, numberOfContributors, isOwner, highlights }: HighlightsPageProps) => {
+  const router = useRouter();
+  const repo = router.query.repo as string;
+
+  const { limit = 10, range = 30 } = router.query;
+  const topRef = useRef<HTMLDivElement>(null);
+
+  const { data: emojis } = useFetchAllEmojis();
+  const { data: taggedRepos } = useListHighlightsTaggedRepos(list?.id ?? "");
+
+  const [contributor, setContributor] = useState("");
+  const debouncedSearchTerm = useDebounceTerm(contributor, 300);
+
+  const { data, isLoading, meta } = useFetchListContributorsHighlights({
+    listId: list?.id ?? "",
+    initialData: highlights,
+    repo,
+    range: range as number,
+    contributor,
+    limit: limit as number,
+  });
+
+  function onSearch(searchTerm: string) {
+    if (!searchTerm || searchTerm.length >= 3) {
+      setContributor(searchTerm);
+    }
+  }
+
+  useEffect(() => {
+    onSearch(contributor);
+  }, [debouncedSearchTerm]);
+
+  return (
+    <ListPageLayout showRangeFilter={false} list={list} numberOfContributors={numberOfContributors} isOwner={isOwner}>
+      <div
+        ref={topRef}
+        className="container flex flex-col justify-start w-full gap-12 pt-4 mt-5 md:mt-0 md:items-start md:flex-row"
+      >
+        <div className={`sticky top-14 xl:flex hidden flex-none w-1/5`}>
+          <div className="w-full bg-white border shadow-sm rounded-lg px-5 py-6 flex flex-col gap-4">
+            <Search
+              value={contributor}
+              onChange={(value) => {
+                onSearch(value);
+              }}
+              placeholder="Search contributors"
+              className="!w-full"
+              name="helo"
+            />
+
+            {taggedRepos && taggedRepos.length > 0 ? (
+              <div className="flex flex-col gap-2">
+                <span className="text-sm text-light-slate-9">Tagged Repositories</span>
+                <SingleSelect
```

Comment:
The `<SingleSelect />` component needs a focus state. I'm going to look at adding this to the global styles as we need it elsewhere.

![CleanShot 2023-12-12 at 16 43 02](https://github.com/open-sauced/app/assets/833231/d9818899-c964-4cec-ad35-d63649a560fa)

Reply:
This work is merged for focus states. #2319